versions/3.1.1.md

```diff
@@ -2762,9 +2766,9 @@ MyResponseType:
       monster: 'https://gigantic-server.com/schemas/Monster/schema.json'
 ```
 
-Here the discriminator _value_ of `dog` will map to the schema `#/components/schemas/Dog`, rather than the default (implicit) value of `Dog`.  If the discriminator _value_ does not match an implicit or explicit mapping, no schema can be determined and validation SHOULD fail. Mapping keys MUST be string values, but tooling MAY convert response values to strings for comparison.
+Here the discriminator _value_ of `dog` will map to the schema `#/components/schemas/Dog`, rather than the default (implicit) `#/components/schemas/dog`.  If the discriminator _value_ does not match an implicit or explicit mapping, no schema can be determined and validation SHOULD fail. Mapping keys MUST be string values, but tooling MAY convert response values to strings for comparison.
```

Comment:
Doesn't this line now contradict the constraint added a few lines above?

> The `discriminator` keyword can not change the validation result of the `oneOf`, it can only help make the evaluation more efficient and provide better error messaging.

Reply:
In this case the validation result of the `discriminator` keyword is "fail". The validation result of the `oneOf` keyword is not affected. The validation result of the schema as whole is affected by this behavior, but not the validation result of the `oneOf` keyword.

In #2621, I remove this constraint because it doesn't seem necessary to cause validation to fail for a keyword that doesn't otherwise affect the validation result. I couldn't make that change here to avoid backward incompatible changes since this is a patch release.